### DISCUSSION TITLE
My ESP8266 don´t go into DeepSleep?!? Is there a foulty in my sketch? - Solved

### DISCUSSION THREAD

**Discussion by Mordi1984** [arendst/Tasmota#16774] [Q&A] [answered]
Hello,

i plan to make an Window Contact and therefor i Use an ESP-12 connected with an Power Regulator to stable 3.1Volt and an some resistors for voltage measurement on ADV Pin.

I bought an Reed Switch with 3 Pins to change the Connection to go into deep sleep, and and this reed is also connected to an other pin and report if the magnet is far away and prevent deep sleep when the switch opened.

But in my Case the ESP-12 is not go into deep sleep again if i set GPIO4 for prevent deep sleep, when opened.
Well if the reed is open i can measure about 7,1 MOhms between ground an GPIO4.
If it is closed with the Magnet i can Measure about 0.1 Ohm.

But why my ESP-12 is not go into deep sleep with GPIO4 closed to Ground, and discharge my battery in about 2 Days?

Do i misunderstand the Manual wrong?
https://tasmota.github.io/docs/DeepSleep/

I changed DeepsleepTime from 0 to 600 and also from 600 to 0 and it will not go into sleep Mode.
If i Change the GPIO4 to none instead of Deep Sleep, he sleeps, and sleep cannot get stopped with GPIO4.

My GPIO Config is:
![image](https://user-images.githubusercontent.com/47158433/194769854-7afc89bc-cece-4c71-9207-4d8ad6d8d335.png)

With Magnet Away:
![image](https://user-images.githubusercontent.com/47158433/194769976-cbb783f9-4e9d-479e-a5b9-6296edd7a99a.png)

With Magnet close to Reed (I was moved it to the reed, thats why it opened and closed):

![image](https://user-images.githubusercontent.com/47158433/194770033-2f59aaae-0840-48b7-b293-646e7cba1125.png)

My Sketch made by Fritzing:
![image](https://user-images.githubusercontent.com/47158433/194770122-a2b1e836-e45e-4d35-af38-6dd031f64152.png)

Hopefully you can Help my with my Plan to let the AA-Batteries not drain down in 2 Days.
I want that the Batteries can be in that Device for more than 2 Years.

Kind regards,

Mordi

**Comment by Mordi1984** [comment, marked as answer]
I Found the Problem, Stupid me, i need to change Deepsleeptime to about 120 again, and the Switch must be open to enable deep sleep again.
If it is closed, he will stay awake till the switch is open again, after an time.

If this is true close the topic, please :)

**Reply by barbudor** [reply]
Discussions doesn't need to be closed

**Reply by barbudor** [reply]
Note that your description of GPIO4 "DeepSleep" is opposite.
When GPIO4 "DeepSleep" is tied to grounf, the ESP will NOT go into DeepSleep
